src/python/build_management/build_manager.py

```diff
@@ -1094,6 +1093,24 @@ def get_build_urls_list(bucket_path, reverse=True):
   return _sort_build_urls_by_revision(build_urls, bucket_path, reverse)
 
 
+def get_primary_bucket_path():
+  """Get the main bucket path for the current job."""
+  release_build_bucket_path = environment.get_value('RELEASE_BUILD_BUCKET_PATH')
+  if release_build_bucket_path:
+    return release_build_bucket_path
+
+  fuzz_target_build_bucket_path = environment.get_value(
+      'FUZZ_TARGET_BUILD_BUCKET_PATH')
+  if fuzz_target_build_bucket_path:
+    fuzz_target = environment.get_value('FUZZ_TARGET')
+    if not fuzz_target:
+      raise BuildManagerException('FUZZ_TARGET is not defined.')
+
+    return fuzz_target_build_bucket_path.replace('%TARGET%', fuzz_target)
+
+  raise BuildManagerException('No primary bucket path defined.')
```

Comment:
You mean like this?

```python
  release_build_bucket_path = environment.get_value('RELEASE_BUILD_BUCKET_PATH')
  if release_build_bucket_path:
    return release_build_bucket_path

  fuzz_target_build_bucket_path = environment.get_value(
      'FUZZ_TARGET_BUILD_BUCKET_PATH')
  if not fuzz_target_build_bucket_path:
    raise BuildManagerException('No primary bucket path defined.')
    
  fuzz_target = environment.get_value('FUZZ_TARGET')
  if not fuzz_target:
    raise BuildManagerException('FUZZ_TARGET is not defined.')

  return fuzz_target_build_bucket_path.replace('%TARGET%', fuzz_target)

```

I feel like in this case avoiding the extra indents makes the logic a little less clear.

Reply:
Yeah, this looks more straightforward to me, but it's not a big deal, up to you :)

Reply:
I am ok with either (like current one slightly better since it give correlation for fuzz target branch). i just want BuildManagerException on more informative on RELEASE_BUILD_BUCKET_PATH and FUZZ_TARGET_BUILD_BUCKET_PATH

Reply:
could you clarify where more info could be added to the exception messages?

Reply:
I mean ". Please define RELEASE_BUILD_BUCKET_PATH or FUZZ_TARGET_BUILD_BUCKET_PATH in the job definition."

Reply:
Done!